include/boost/json/impl/array.ipp

```diff
@@ -754,4 +755,27 @@ equal(
 
 BOOST_JSON_NS_END
 
+//----------------------------------------------------------
+//
+// std::hash specialization
+//
+//----------------------------------------------------------
+
+namespace std {
+std::size_t 
+hash<::boost::json::array>::operator()(
+    ::boost::json::array const& ja) const noexcept 
+{
+  std::size_t seed = ja.size();
+  for (const auto& jv : ja) {
+    seed = boost::json::detail::hash_combine(
```

Comment:
`::boost::json::detail::hash_combine` ?

Reply:
I don't think there's any need to open `namespace std` here.

Reply:
Once a user had reported me a GCC bug (affecting from GCC4.8 to 7) related to such specializations:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56480

The quick solution for that is to open the `namespace std`.

Reply:
That's not the same. You're defining the member function here, not specializing the class template.

Reply:
Done.